### PR TITLE
add support for compressed responses

### PR DESCRIPTION
**Pull request by thehesiod**
- resolves https://github.com/aio-libs/aiobotocore/issues/295 + https://github.com/aio-libs/aiobotocore/issues/258
- relies on https://github.com/aio-libs/aiohttp/pull/2110

### CODE REVIEW

**Comment by jettify** [comment]
`I just noticed that if I try a aiobotocore s3 client get_object of a key that a Content-Encoding metadata set to gzip, it's automatically decompressing the body as well, not-good...`

Good catch! Looks like `aiobotocore` hits every edge case of aiohttp client possible.

**Comment by thehesiod** [comment]
crazy eh? :)  Also my service seems to have found a python segfault after updating a defaultdict a few million times <sigh>  that's not going to be fun to track down.

**Comment by thehesiod** [comment]
oh, and in reality the issue with aiohttp is that it doesn't follow http stack norms :(   Basically designed like: I think it should behave like X by default, whereas every other library behaves like Y

**Comment by jettify** [comment]
aio-libs/aiohttp#2110 had been merged, do you have idea how we can test this PR?

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 # aiohttp requirement is pegged as we need to manually ensure that
 # https://github.com/aio-libs/aiobotocore/pull/248 will continue working
-install_requires = ['botocore>=1.5.71, <=1.5.78', 'aiohttp>=2.0.4, <=2.3.0',
+install_requires = ['botocore>=1.5.71, <=1.5.78', 'aiohttp>=2.X.X, <=2.X.X',
```
Comment (jettify) [inline setup.py]: updates her should fix CI

**Comment by thehesiod** [comment]
it's been merged but there's no release available with it yet

File: setup.py (hunk 1 above)
Comment (thehesiod) [reply setup.py]: ya, broken on purpose for now so we set the right version that's needed

Comment (jettify) [reply setup.py]: got it

**Comment by jettify** [comment]
merged manually here https://github.com/aio-libs/aiobotocore/pull/425

**Pull request closed without merging**